pkg/controller/accountclaim/fake.go

```diff
@@ -0,0 +1,83 @@
+package accountclaim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	controllerutils "github.com/openshift/aws-account-operator/pkg/controller/utils"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ReconcileAccountClaim) processFake(reqLogger logr.Logger, accountClaim *awsv1alpha1.AccountClaim) (bool, error) {
```

Comment:
I think it would be very helpful to have a comment describing this method and what "fake" means in the context of processing/reconciling an `AccountClaim`. What are the properties of a "fake," what is the lifecycle of a "fake," how are they used?

The word "fake" is used throughout and that has a context in testing (at least to this reviewer) to be something that does not exist. Do these resources actually exist in a cluster for the purposes of supporting external tests? Later, when "fake" resources are deleted, is that making a change inside a cluster? Or is all of this taking place inside the aws-account-operator's memory?

Reply:
Would be good to add a unit test covering this function